data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetBlobSidecars.java

```diff
@@ -86,21 +90,26 @@ public void handleRequest(final RestApiRequest request) throws JsonProcessingExc
                     .orElse(AsyncApiResponse.respondNotFound())));
   }
 
-  private static SerializableTypeDefinition<List<BlobSidecar>> getResponseType(
+  private static SerializableTypeDefinition<BlobSidecarsAndMetaData> getResponseType(
       final SchemaDefinitionCache schemaCache) {
     final DeserializableTypeDefinition<BlobSidecar> blobSidecarType =
         SchemaDefinitionsDeneb.required(schemaCache.getSchemaDefinition(SpecMilestone.DENEB))
             .getBlobSidecarSchema()
             .getJsonTypeDefinition();
-    return SerializableTypeDefinition.<List<BlobSidecar>>object()
+    return SerializableTypeDefinition.<BlobSidecarsAndMetaData>object()
         .name("GetBlobSidecarsResponse")
-        .withField("data", listOf(blobSidecarType), Function.identity())
+        .withOptionalField("version", MILESTONE_TYPE, (b) -> Optional.of(b.getMilestone()))
```

Comment:
The OpenAPI spec has defined them as optional. So I believe we need to use `withOptionalField()` to ensure we are compliant, don't we?

Reply:
Yes, if we want to deserialize as well. Otherwise, not needed, since we always add them, but not fuzzed about it. Probably defining them as optional makes it more clear.

Reply:
I was thinking from the perspective of the open api spec that we generate. If we want to match the "official" spec, we would need the optional field so we would generate the open api schema spec as optional as well.

Reply:
Yeah, that makes sense.